src/Symfony/Component/Config/Builder/ConfigBuilderGenerator.php

```diff
@@ -172,17 +172,18 @@ private function handlePrototypedArrayNode(PrototypedArrayNode $node, ClassBuild
         if (null !== $parameterType || $prototype instanceof ScalarNode) {
             $property = $class->addProperty($node->getName());
             if (null === $key = $node->getKeyAttribute()) {
+                // This is a array of scalar values. Dont use "singular name".
```

Comment:
this is an array of values. Maybe not scalar (see VariableNode, which allows to put non-scalar values treated the same than scalar values)